File: crates/analyzer/general/tests/corpus/bad/handlers.js
// EXPECT: gen_xss_sink
// EXPECT: gen_command_injection
// EXPECT: gen_sql_injection
// EXPECT: gen_insecure_randomness
const { exec } = require("child_process");

function show(el, req) {
  el.innerHTML = req.query.name;
}
function ping(req) {
  exec(`ping -c 1 ${req.query.host}`);
}
function find(db, req) {
  return db.query(`SELECT * FROM users WHERE id = ${req.params.id}`);
}
function resetToken() {
  return Math.random().toString(36).slice(2);
}

File: crates/analyzer/general/tests/corpus/bad/handlers.py
# EXPECT: gen_command_injection
# EXPECT: gen_sql_injection
# EXPECT: gen_unsafe_deserialization
# EXPECT: gen_tls_verification_disabled
import subprocess, pickle, requests

def ping(request, conn):
    host = request.args["host"]
    subprocess.run(f"ping -c 1 {host}", shell=True)
    cur = conn.cursor()
    cur.execute("SELECT * FROM users WHERE name = '%s'" % request.args["name"])
    state = pickle.loads(request.data)
    return requests.get(request.args["url"], verify=False)
